js/components/frame.js

```diff
@@ -764,7 +764,7 @@ class Frame extends ImmutableComponent {
         const runInsecureContent = parsedUrl.protocol === 'https:' && this.runInsecureContent()
         windowActions.setSecurityState(this.frame, {
           secure: isSecure,
-          runInsecureContent: runInsecureContent
+          runInsecureContent: runInsecureContent ? this.props.location : null
```

Comment:
why does location need to be in the securityState? `SiteInfo` can just do `frameProps.get('location')`